### PR TITLE
Test Emacs parts on the released 28.1 as well

### PR DESCRIPTION
**Pull request by mattiase**
28.1 was recently released; `snapshot` tracks the development version (with some lag).
Oh, and `28.1` in YAML is a number so I prefer the quotes to make it a string just in case.

### CODE REVIEW

**Comment by voodoos** [comment]
It is worth noting that the lint erros don't trigger CI failures. It was meant to have the log readily available for when we want to fix them:
```

merlin-ac.el:1:0: warning: The package summary should not end with a period.
merlin-ac.el:135:0: error: "ac-merlin-locate" doesn't start with package's prefix "merlin-ac".
merlin-cap.el:1:0: error: Package should have a ;;; Commentary section.
merlin-cap.el:1:0: warning: The package summary should not end with a period.
merlin-cap.el:1:87: warning: You should depend on (emacs "24.1") if you need lexical-binding.
merlin-cap.el:21:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
merlin-cap.el:24:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
merlin-cap.el:72:0: error: Aliases should start with the package's prefix "merlin-cap".
merlin-company.el:1:0: warning: The package summary should not end with a period.
merlin-iedit.el:1:0: error: package.el cannot parse this buffer: Search failed: ";;; merlin-iedit.el ends here"
merlin-imenu.el:1:0: error: Package should have a ;;; Commentary section.
merlin-imenu.el:1:0: warning: The package summary should not end with a period.
merlin-imenu.el:1:75: warning: You should depend on (emacs "24.1") if you need lexical-binding.
merlin-imenu.el:10:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
merlin-imenu.el:11:7: error: Package URLs should be a single HTTPS or HTTP URL.
merlin-imenu.el:14:10: error: You should depend on (emacs "24.4") if you need `subr-x'.
merlin-imenu.el:18:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
merlin-imenu.el:19:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
merlin-imenu.el:20:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
merlin-imenu.el:21:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
merlin-imenu.el:22:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
merlin-imenu.el:23:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
merlin-imenu.el:102:0: error: "merlin-use-merlin-imenu" doesn't start with package's prefix "merlin-imenu".
merlin-xref.el:1:0: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
merlin-xref.el:1:0: error: Package should have a ;;; Commentary section.
merlin-xref.el:1:0: error: Package should have a Homepage or URL header.
merlin-xref.el:1:0: error: package.el cannot parse this buffer: Package lacks a file header
merlin-xref.el:1:7: warning: You should depend on (emacs "24.1") if you need lexical-binding.
merlin-xref.el:2:10: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-lib'.
merlin-xref.el:3:10: error: You should depend on (emacs "25.1") if you need `xref'.
merlin-xref.el:16:1: error: You should depend on (emacs "25.1") or the cl-lib package if you need `cl-defmethod'.
merlin-xref.el:19:35: error: You should depend on (emacs "25.1") if you need `alist-get'.
merlin-xref.el:20:8: error: You should depend on (emacs "25.1") if you need `xref-make'.
merlin-xref.el:21:19: error: You should depend on (emacs "25.1") if you need `xref-make-buffer-location'.
merlin-xref.el:24:1: error: You should depend on (emacs "25.1") or the cl-lib package if you need `cl-defmethod'.
merlin-xref.el:26:16: error: You should depend on (emacs "25.1") if you need `alist-get'.
merlin-xref.el:27:15: error: You should depend on (emacs "25.1") if you need `alist-get'.
merlin-xref.el:28:16: error: You should depend on (emacs "25.1") if you need `alist-get'.
merlin-xref.el:29:15: error: You should depend on (emacs "25.1") if you need `alist-get'.
merlin-xref.el:33:15: error: You should depend on (emacs "25.1") if you need `xref-make'.
merlin-xref.el:33:31: error: You should depend on (emacs "25.1") if you need `xref-make-file-location'.
merlin-xref.el:35:1: error: You should depend on (emacs "25.1") or the cl-lib package if you need `cl-defmethod'.
```

We should definitely address them sometimes and then make the lint failures also fail the CI.

**Comment by mattiase** [comment]
The `check.sh` machinery never really worked properly but with these changes it should (I had to look up the shell parameter expansion rules for the umpteenth time).

There are still package-lint complaints but they seem to be difficult to chase away, especially when doing something nonstandard out of necessity. At least byte-compile warnings and package-lint are now controlled by different levers.

File: emacs/merlin.el [hunk 1]
```diff
@@ -1637,7 +1637,7 @@ Empty string defaults to jumping to all these."
   (merlin--goto-file-and-point (merlin-call-jump target)))
 
 (defun merlin-call-phrase (target)
-  "Move to next phrase (TARGET = 'next) or previous phrase (TARGET = 'prev)"
+  "Move to next phrase (TARGET = `next') or previous phrase (TARGET = `prev')."
```
Comment (bbatsov) [inline emacs/merlin.el:1640]: I wouldn't be using quote here at all, as I think they actually make things more confusing.

File: emacs/merlin.el [hunk 2]
```diff
@@ -1813,7 +1813,7 @@ Empty string defaults to jumping to all these."
 ;;;;;;;;;;;;;;;;;;;
 
 (defun merlin--refactor-open (mode)
-  "Refactor open statement under cursor. mode can be 'qualify or 'unqualify"
+  "Refactor open statement under cursor. mode can be `qualify' or `unqualify'."
```
Comment (bbatsov) [inline emacs/merlin.el:1816]: Same here.

**Comment by mattiase** [comment]
As this PR has lingered unapplied, Emacs 29 has gained new warnings. The latest update takes care of them all (and fixes some doc string quoting mistakes that I overlooked the first time).

**Review by voodoos — APPROVED** [review]

**Pull request merged**